### PR TITLE
Fix writing marks in StorageLog

### PR DESCRIPTION
**Pull request by vitlibar**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category:
- Improvement

Changelog entry:
This PR allows reading data of a table using the `Log` engine in multiple threads.

### CODE REVIEW

File: src/Storages/StorageLog.cpp [hunk 1]
```diff
@@ -214,8 +214,7 @@ class LogSink final : public SinkToStorage
         , storage(storage_)
         , metadata_snapshot(metadata_snapshot_)
         , lock(std::move(lock_))
-        , marks_stream(
-            storage.disk->writeFile(storage.marks_file_path, 4096, WriteMode::Rewrite))
+        , marks_stream(storage.disk->writeFile(storage.marks_file_path, 4096, WriteMode::Append))
```
Comment (vitlibar) [inline src/Storages/StorageLog.cpp:217]: It seems new marks should be appended to the older marks, not to replace them. And originally here was written `APPEND`, not `REWRITE`, but it seems https://github.com/ClickHouse/ClickHouse/pull/8862/files#diff-42d2ff7e53241a29f266116cea5661960ed5d771bc271066778df3c1c1a19e04R112 changed that.

**Review by kitaisreal — APPROVED** [review]

**Comment by azat** [comment]
>It's not obvious how to write a good test from user's perspective

AFAICS after this patch the Log table can be read from multiple threads if it has > 1 marks, right? So I guess it can be tested via system.query_log.thread_ids (with this patch it will use more threads).

**Comment by alexey-milovidov** [comment]
Ok. Also if adding test is cumbersome - it's ok without a test.

**Comment by azat** [comment]
>Integration tests (release) — fail: 2, passed: 1596, flaky: 0

- #30150

**Pull request merged**